sentry/sentry.go

```diff
@@ -0,0 +1,212 @@
+		client, err := raven.New(dsn)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create sentry client")
+		}
+
+		l.Capturer = &NonBlockingCapturer{Client: client}
+	}
+
+	return l, nil
+}
+
+// WithCapturer allows to specify which packet capturer should be used
+func WithCapturer(c Capturer) func(l *Logger) {
+	return func(l *Logger) {
+		l.Capturer = c
+	}
+}
+
+// WithMinLevel provides a slice of levels for which sentry packaet should be sent
+func WithMinLevel(level zap.Level) func(l *Logger) {
+	return func(l *Logger) {
+		l.minLevel = level
+	}
+}
+
+// WithTraceEnabled allows to change wether (the somewhat expensive) stack trace generation
+// takes place
+func WithTraceEnabled(enabled bool) func(l *Logger) {
+	return func(l *Logger) {
+		l.traceEnabled = enabled
+	}
+}
+
+// WithTraceCfg allows to change the number of skipped frames, number of context lines and
+// list of go prefixes that are considered "in-app", i.e. "github.com/uber-go/zap"
+func WithTraceCfg(skip, context int, prefixes []string) func(l *Logger) {
+	return func(l *Logger) {
+		l.traceSkipFrames = skip
+		l.traceContextLines = context
+		l.traceAppPrefixes = prefixes
+	}
+}
+
+// WithExtra stores additional information for each Sentry event
+func WithExtra(extra map[string]interface{}) func(l *Logger) {
+	return func(l *Logger) {
+		l.extra = extra
+	}
+}
+
+// Log sends Sentry information provided minimum threshold is met
+func (l *Logger) Log(lvl zap.Level, msg string, fields ...zap.Field) {
+	l.log(lvl, msg, fields)
+}
+
+// Debug is a nop
+func (l *Logger) Debug(msg string, fields ...zap.Field) {}
+
+// Info sends Sentry information provided minimum threshold is met
+func (l *Logger) Info(msg string, fields ...zap.Field) {
+	l.log(zap.InfoLevel, msg, fields)
+}
+
+// Warn sends Sentry information provided minimum threshold is met
+func (l *Logger) Warn(msg string, fields ...zap.Field) {
+	l.log(zap.WarnLevel, msg, fields)
+}
+
+// Error sends Sentry information provided minimum threshold is met
+func (l *Logger) Error(msg string, fields ...zap.Field) {
+	l.log(zap.ErrorLevel, msg, fields)
+}
+
+// Panic sends Sentry information provided minimum threshold is met
+func (l *Logger) Panic(msg string, fields ...zap.Field) {
+	l.log(zap.PanicLevel, msg, fields)
+}
+
+// Fatal sends Sentry information provided minimum threshold is met
+func (l *Logger) Fatal(msg string, fields ...zap.Field) {
+	l.log(zap.FatalLevel, msg, fields)
+}
+
+// DFatal is a nop
+func (l *Logger) DFatal(msg string, fields ...zap.Field) {}
+
+// With returns Sentry logger with additional context
+func (l *Logger) With(fields ...zap.Field) zap.Logger {
+	// Consider pre-allocing the whole new resulting map, but that may be even more expensive
+	for _, f := range fields {
+		f.AddTo(l.extra)
+	}
+	return l
+}
+
+// Notify sentry if the log level meets minimum threshold
+func (l *Logger) log(lvl zap.Level, msg string, fields []zap.Field) {
+	if lvl >= l.minLevel {
```

Comment:
Should be using `Meta.Enabled`, if we could do away with the `minLevel` field here and just use `Meta`'s leveling support